server/channel.js

```diff
@@ -0,0 +1,72 @@
+var WebSocketServer = require("ws").Server;
+var fs              = require("fs");
+
+// this channel.js is the server-side abstraction over websockets
```

Comment:
There's a lot of duplicate code here with the other channel.js file. Can you find a way to share them and use the same channel abstraction on both client and server?